fs/memoryfs.py

```diff
@@ -207,6 +207,44 @@ def __init__(self, resource_type, name):
         if not self.is_dir:
             self._bytes_file = io.BytesIO()
 
+    def clean_up(self, force=False):
+        # type: (bool) -> None
+        """Release resources held by this directory entry.
+
+        Arguments:
+            force (bool):
+                Force remaining open files to close and delete subdirectory entries.
+
+        Raises:
+            fs.errors.ResourceLocked:
+                This directory entry has remaining open files and ``force`` is `False`.
+            fs.errors.DirectoryNotEmpty:
+                This directory has remaining entries and ``force`` is `False`.
+        """
+        with self.lock:
+            if self._open_files and not force:
+                raise errors.ResourceLocked(
```

Comment:
The `close()` method shouldn't throw exceptions. It's not an operation that can fail as such.